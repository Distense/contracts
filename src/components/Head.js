import React from 'react'
import { Helmet } from 'react-helmet'

export default ({ title, children }) => (
  <Helmet>
    <title>{title ? `${title} - Distense` : 'Distense'}</title>
    {children}
  </Helmet>
)
