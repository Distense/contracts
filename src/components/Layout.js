import React from 'react'
import { Link } from 'react-router-dom'
import { Container, Menu, Segment } from 'semantic-ui-react'

import EthereumAuthenticated from './common/EthereumAuthenticated'
// import GasEstimate from './common/GasEstimate'
import NumberDIDOwned from './common/NumberDIDOwned'
import HasWeb3 from './common/HasWeb3'
import TotalDID from './common/TotalDID'

import Status from './Status'

export default ({ children, title }) => (
  <div>
    <Segment
      inverted
      textAlign="center"
      style={{
        padding: '.8em 0em'
      }}
      vertical
    >
      <Menu
        borderless
        fixed="top"
        className="inconsolata"
        inverted
        size="large"
      >
        <Container textAlign="center">
          <Menu.Item to="/" as={Link} position="left">
            Distense
          </Menu.Item>
          <Menu.Item to="/tasks/add" as={Link}>
            Propose
          </Menu.Item>
          <Menu.Item to="/tasks" as={Link}>
            View
          </Menu.Item>
          <Menu.Item to="/pullrequests/add" as={Link}>
            Submit
          </Menu.Item>
          <Menu.Item to="/pullrequests" as={Link}>
            Approve
          </Menu.Item>
          <Menu.Item to="/exchange" as={Link}>
            Exchange
          </Menu.Item>
          <TotalDID />
        </Container>
      </Menu>
    </Segment>

    <Container style={{ marginTop: '4em' }}>{children}</Container>

    <Segment vertical style={{ margin: '3em 0em 0em', padding: '5em 0em' }}>
      <Menu className="inconsolata" borderless fixed="bottom">
        <Container>
          <Menu.Item>Â© {new Date().getFullYear()} Distense</Menu.Item>
          <Menu.Item
            style={{ textDecoration: 'underline' }}
            color="blue"
            link
            as={Link}
            position="right"
            to="/FAQ"
          >
            FAQ
          </Menu.Item>
          <Menu.Item
            style={{ textDecoration: 'underline' }}
            color="blue"
            link
            as={Link}
            to="/parameters"
          >
            Parameters
          </Menu.Item>
          <Menu.Item
            style={{ textDecoration: 'underline' }}
            color="blue"
            link
            as={Link}
            to="/events"
          >
            Event Feed
          </Menu.Item>
          <Status />
          <NumberDIDOwned />
          {/*<GasEstimate />*/}
          <HasWeb3 />
          <EthereumAuthenticated />
        </Container>
      </Menu>
    </Segment>
    {/*language=CSS*/}
    <style global jsx>{`
      .inconsolata {
        font-family: 'Inconsolata', sans-serif !important;
      }
    `}</style>
  </div>
)
