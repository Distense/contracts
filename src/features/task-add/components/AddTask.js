import React, { Component } from 'react'
import Autosuggest from 'react-autosuggest'

import { Link } from 'react-router-dom'
import {
  Button,
  Dropdown,
  Input,
  Form,
  Grid,
  List,
  Message
} from 'semantic-ui-react'
import { Redirect } from 'react-router'

import Head from '../../../components/Head'
import PageTitling from '../../../components/PageTitling'
import { tagsOptions } from '../../tasks/operations/tagsOptions'
import {
  getSuggestions,
  getSuggestionValue,
  renderSuggestion
} from '../autosuggestHelpers'

export default class AddTask extends Component {
  constructor(props) {
    super(props)
    this.state = {
      value: '',
      tagsString: '',
      issueNum: '',
      issues: [],
      repo: '',
      redirect: false
    }

    this.onChangeRepo = this.onChangeRepo.bind(this)
    this.onChangeTags = this.onChangeTags.bind(this)
  }

  onChangeTitle = (event, { newValue }) => {
    //  make sure not to have any slashes for future URLs
    if (newValue) newValue = newValue.replace('/', '-')
    this.setState({ value: newValue })
  }

  onChangeIssueNum = ({ target: { value } }) => {
    this.setState({ issueNum: value })
  }

  onChangeTags(e, data) {
    const tags = data.value
    if (tags.length < 3) {
      this.setState({ tags })
      let tagsString = ''
      tags.forEach((tag, i) => {
        if (i === 0) tagsString = tag
        else tagsString += ':' + tag
      })
      this.setState({ tagsString })
    }
  }

  onChangeRepo(e, data) {
    this.setState({
      repo: data.value
    })
  }

  onSubmit = async e => {
    e.preventDefault()

    const { title, tagsString, issueNum, repo } = this.state

    if (title && tagsString && issueNum && repo) {
      this.props.addTask({ title, tagsString, issueNum, repo })
      this.setState({
        redirect: true
      })
    }
  }

  onSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      issues: getSuggestions(this.props.issues, value)
    })
  }

  onSuggestionsClearRequested = () => {
    this.setState({
      issues: []
    })
  }

  render() {
    const {
      issueNum,
      issues,
      loading,
      redirect,
      repo,
      value,
      tags
    } = this.state

    const { numDIDRequiredToAddTask } = this.props

    if (redirect) return <Redirect to="/tasks" />

    if (loading) return <p>Loading ...</p>

    const titleProps = {
      placeholder: `Select from existing issue titles and we'll handle the rest`,
      onChange: this.onChangeTitle,
      value
    }

    return (
      <div>
        <Head title="Add Task" />
        <Grid>
          <PageTitling
            title="Propose"
            subtitle="DID holders can propose new work"
          />
          <Grid.Row style={{ paddingTop: '0px' }} columns={2}>
            <Grid.Column>
              <Form onSubmit={this.onSubmit}>
                <Form.Field>
                  <Autosuggest
                    alwaysRenderSuggestions={true}
                    suggestions={issues}
                    getSuggestionValue={getSuggestionValue}
                    renderSuggestion={renderSuggestion}
                    onSuggestionsFetchRequested={
                      this.onSuggestionsFetchRequested
                    }
                    inputProps={titleProps}
                    onSuggestionsClearRequested={
                      this.onSuggestionsClearRequested
                    }
                    highlightFirstSuggestion={true}
                  />
                </Form.Field>
                <Form.Field>
                  <Input
                    placeholder="Tags"
                    onChange={this.onChangeIssueNum}
                    value={issueNum}
                  />
                </Form.Field>
                <Button size="large" color="green" type="submit">
                  Submit
                </Button>
              </Form>
            </Grid.Column>
            <Grid.Column width={8}>
              <Form>
                <Form.Field>
                  <Dropdown
                    fluid
                    multiple
                    onChange={this.onChangeTags}
                    options={tagsOptions}
                    placeholder="Tags"
                    search
                    selection
                    scrolling
                    value={tags}
                  />
                </Form.Field>
                <Form.Field>
                  <Dropdown
                    fluid
                    onChange={this.onChangeRepo}
                    options={[
                      {
                        key: 'ui',
                        value: 'ui',
                        text: 'UI'
                      },
                      {
                        key: 'contracts',
                        value: 'contracts',
                        text: 'Contracts'
                      }
                    ]}
                    placeholder="Repo"
                    search
                    selection
                    scrolling
                    value={repo ? repo : 'ui'}
                  />
                </Form.Field>
              </Form>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={16}>
              <Message>
                <Message.Header>Rules</Message.Header>
                <List bulleted>
                  <List.Item>
                    You must own at least {numDIDRequiredToAddTask} DID to
                    propose. This number changes according to the{' '}
                    <Link to="/parameters">proposalPctDIDToApprove</Link>{' '}
                    parameter
                  </List.Item>
                  <List.Item>
                    Your proposal can be anything, it doesn't necessarily have
                    to be <em>work</em>.
                  </List.Item>
                  <List.Item>
                    Remember that when you propose, it is likely that DID will
                    be issued for the completion of the task. The fewer DID the
                    better.
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Github Issue Number</Message.Header>
                <List bulleted>
                  <List.Item>
                    Create an issue in the appropriate Github repository to
                    determine the Github issue number
                  </List.Item>
                  <List.Item>
                    Your proposal can be anything, it doesn't necessarily have
                    to be <em>work</em>.
                  </List.Item>
                  <List.Item>
                    Discussion will happen on Github in the issue you create
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Title</Message.Header>
                <List bulleted>
                  <List.Item>Enter the title of your task/proposal</List.Item>
                  <List.Item>
                    Limit the length of the title: the more characters the more
                    gas it will cost you
                  </List.Item>
                  <List.Item>
                    Example: 'Build this amazing new dapp that will change the
                    world'
                  </List.Item>
                  <List.Item>
                    People will like you more if you enter the same exact title
                    as your Github issue
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Tags</Message.Header>
                <List bulleted>
                  <List.Item>
                    Select the category of proposal you're making
                  </List.Item>
                  <List.Item>
                    Examples are: 'contracts', 'frontend', 'governance'
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Repo</Message.Header>
                <List bulleted>
                  <List.Item>Choose the one that best fits</List.Item>
                </List>
              </Message>
            </Grid.Column>
          </Grid.Row>
        </Grid>
        {/*language=CSS*/}
        <style global jsx>{`
          .react-autosuggest__suggestions-list {
            list-style-type: none;
          }

          .ui.form {
            font-size: 1.25rem;
          }

          .react-autosuggest__suggestion {
            border: 1px solid gray;
            border-radius: 3px 3px 3px 3px;
            -moz-border-radius: 3px 3px 3px 3px;
            -webkit-border-radius: 3px 3px 3px 3px;
            /*color: #888888;*/
            line-height: 2.9rem;
            margin-left: -38px;
            padding: 2px 10px;
            font-size: large;
          }
        `}</style>
      </div>
    )
  }
}
