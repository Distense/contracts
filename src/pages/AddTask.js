import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import {
  Button,
  Dropdown,
  Input,
  Form,
  Grid,
  List,
  Message
} from 'semantic-ui-react'
import { connect } from 'react-redux'
import { Redirect } from 'react-router'

import { addTask } from '../actions/tasks'
import { getPendingTask } from '../reducers/tasks'
import { NUM_DID_REQUIRED_TO_ADD_TASK_PARAMETER_TITLE } from '../constants/parameters/parameterTitles'

import Head from '../components/common/Head'
import PageTitling from '../components/common/PageTitling'
import { tagsOptions } from '../tagsOptions'
import { getParameterValueByTitle } from '../reducers/parameters'

export class AddTask extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: '',
      tagsString: '',
      issueNum: '',
      numDIDRequiredToAddTask: this.props.numDIDRequiredToAddTask || 120,
      repo: '',
      redirect: false
    }

    this.onChangeRepo = this.onChangeRepo.bind(this)
    this.onChangeTags = this.onChangeTags.bind(this)
  }

  componentDidMount() {
    this.someTimeout = setTimeout(() => {
      this.setState({
        numDIDRequiredToAddTask: this.props.numDIDRequiredToAddTask
      })
    }, 2000)
  }

  onChangeTitle = ({ target: { value } }) => {
    //  make sure not to have any slashes for URLs
    value = value.replace('/', '')
    if (value.length <= 70) {
      this.setState({ title: value })
    }
  }

  onChangeIssueNum = ({ target: { value } }) => {
    this.setState({ issueNum: value })
  }

  onChangeTags(e, data) {
    const tags = data.value
    if (tags.length < 3) {
      this.setState({ tags })
      let tagsString = ''
      tags.forEach((tag, i) => {
        if (i === 0) tagsString = tag
        else tagsString += ':' + tag
      })
      this.setState({ tagsString })
    }
  }

  onChangeRepo(e, data) {
    this.setState({
      repo: data.value
    })
  }

  onSubmit = async e => {
    e.preventDefault()

    const { title, tagsString, issueNum, repo } = this.state

    this.props.addTask({ title, tagsString, issueNum, repo })
    this.setState({
      redirect: true
    })
  }

  render() {
    const {
      title,
      tags,
      issueNum,
      numDIDRequiredToAddTask,
      repo,
      redirect
    } = this.state

    if (redirect) return <Redirect to="/tasks" />

    return (
      <div>
        <Head title="Add Task" />
        <Grid>
          <Grid.Row>
            <PageTitling
              title="Propose"
              subtitle="DID holders can propose new work here"
            />
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={8}>
              <Form onSubmit={this.onSubmit}>
                <Form.Field>
                  <Input
                    onChange={this.onChangeIssueNum}
                    placeholder="Github Issue Number"
                    value={issueNum}
                  />
                </Form.Field>
                <Form.Field>
                  <Input
                    type="text"
                    placeholder="Title (<70 chars)"
                    onChange={this.onChangeTitle}
                    className=""
                    name="title"
                    value={title}
                  />
                </Form.Field>
                <Button size="large" color="green" type="submit">
                  Submit
                </Button>
              </Form>
            </Grid.Column>
            <Grid.Column width={8}>
              <Form>
                <Form.Field>
                  <Dropdown
                    fluid
                    multiple
                    onChange={this.onChangeTags}
                    options={tagsOptions}
                    placeholder="Tags"
                    search
                    selection
                    scrolling
                    value={tags}
                  />
                </Form.Field>
                <Form.Field>
                  <Dropdown
                    fluid
                    onChange={this.onChangeRepo}
                    options={[
                      {
                        key: 'ui',
                        value: 'ui',
                        text: 'UI'
                      },
                      {
                        key: 'contracts',
                        value: 'contracts',
                        text: 'Contracts'
                      }
                    ]}
                    placeholder="Repo"
                    search
                    selection
                    scrolling
                    value={repo}
                  />
                </Form.Field>
              </Form>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={16}>
              <Message>
                <Message.Header>Rules</Message.Header>
                <List bulleted>
                  <List.Item>
                    You must own at least {numDIDRequiredToAddTask} DID to
                    propose. This number changes according to the{' '}
                    <Link to="/parameters">proposalPctDIDToApprove</Link>{' '}
                    parameter
                  </List.Item>
                  <List.Item>
                    Your proposal can be anything, it doesn't necessarily have
                    to be <em>work</em>.
                  </List.Item>
                  <List.Item>
                    Remember that when you propose, it is likely that DID will
                    be issued for the completion of the task. The fewer DID the
                    better.
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Github Issue Number</Message.Header>
                <List bulleted>
                  <List.Item>
                    Create an issue in the appropriate Github repository to
                    determine the Github issue number
                  </List.Item>
                  <List.Item>
                    Your proposal can be anything, it doesn't necessarily have
                    to be <em>work</em>.
                  </List.Item>
                  <List.Item>
                    Discussion will happen on Github in the issue you create
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Title</Message.Header>
                <List bulleted>
                  <List.Item>Enter the title of your task/proposal</List.Item>
                  <List.Item>
                    Limit the length of the title: the more characters the more
                    gas it will cost you
                  </List.Item>
                  <List.Item>
                    Example: 'Build this amazing new dapp that will change the
                    world'
                  </List.Item>
                  <List.Item>
                    People will like you more if you enter the same exact title
                    as your Github issue
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Tags</Message.Header>
                <List bulleted>
                  <List.Item>
                    Select the category of proposal you're making
                  </List.Item>
                  <List.Item>
                    Examples are: 'contracts', 'frontend', 'governance'
                  </List.Item>
                </List>
              </Message>
            </Grid.Column>
            <Grid.Column width={8}>
              <Message>
                <Message.Header>Repo</Message.Header>
                <List bulleted>
                  <List.Item>Choose the one that best fits</List.Item>
                </List>
              </Message>
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  pendingTask: getPendingTask(state),
  numDIDRequiredToAddTask: getParameterValueByTitle(
    state,
    NUM_DID_REQUIRED_TO_ADD_TASK_PARAMETER_TITLE
  )
})

const mapDispatchToProps = dispatch => ({
  addTask: task => dispatch(addTask(task))
})

export default connect(mapStateToProps, mapDispatchToProps)(AddTask)
