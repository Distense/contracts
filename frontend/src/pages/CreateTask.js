import React, { Component } from 'react'
import CodeMirror from 'react-codemirror2'
import 'codemirror/lib/codemirror.css'
import 'codemirror/mode/markdown/markdown'
import { Dropdown } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { Redirect } from 'react-router'
import ReactMarkdown from 'react-markdown'
import slug from 'slug'

import { getPendingTask } from '../reducers/tasks'
import { createTask } from '../actions'
import Head from '../components/common/Head'
import Layout from '../components/Layout'
import { tagsOptions, specPlaceholder } from '../shared'

const taskUrl = ({ title, _id }) => `/tasks/${slug(title)}/${_id}`


class CreateTask extends Component {
  constructor(props) {
    super(props)
    this.state = {
      title: '',
      tags: [],
      spec: specPlaceholder
    }

    this.onChangeTags = this.onChangeTags.bind(this)
  }

  onChangeTitle = ({ target: { value } }) => {
    if (value.length <= 50)
      this.setState({ title: value })
  }

  onChangeTags(e, data) {
    const tags = data.value
    if (tags.length < 6)
      this.setState({ tags })
  }

  onChangeSpec = (editor, metadata, value) => {
    this.setState({ spec: value })
  }

  onSubmit = async (e) => {
    e.preventDefault()
    const { title, tags, spec } = this.state
    this.props.createTask({ title, tags, spec })
  }

  render() {
    const { pendingTask } = this.props
    const {
      title,
      tags,
      spec
    } = this.state

    if (pendingTask) {
      //  TODO probably redirect to /tasks
      //  This does not allow navigating to /tasks
      //  then navigating to CreateTask while !!pendingTask
      return <Redirect to={taskUrl(pendingTask)}/>
    }

    return (
      <Layout>
        <Head title='Create Task'/>
        <div className='ui one column grid'>
          <form className='ui form' onSubmit={this.onSubmit}>
            <h1 className='ui header'>Create Task</h1>
            <div className='field'>
              <input
                type='text'
                placeholder='Title'
                onChange={this.onChangeTitle}
                className=''
                name='title'
                value={title}
              />
            </div>
            <div className='field'>
              <Dropdown
                // compact
                fluid
                multiple
                onChange={this.onChangeTags}
                options={tagsOptions}
                placeholder='Tags'
                search
                selection
                scrolling
                value={tags}
              />
            </div>
            <div className='fields fields-margin-fix'>
              <div className='eight wide field'>
                <CodeMirror value={spec}
                  options={{
                    cursorBlinkRate: 650,
                    lineNumbers: true,
                    lineWrapping: true,
                    mode: 'markdown',
                    scrollbarStyle: null,
                    tabSize: 2
                  }}
                  onValueChange={this.onChangeSpec}
                />
              </div>
              <div className='eight wide field react-markdown'>
                <ReactMarkdown source={spec} />
              </div>
            </div>
            <button className='ui large green button' type='submit'>
              Submit
            </button>
          </form>
        </div>

        { /*language=CSS*/ }
        <style global jsx>{`

          .react-codemirror2, .react-markdown {
            border: 1px solid rgba(34,36,38,.15);
            border-radius: .28571429rem;
          }

          .react-codemirror2 {
            border-top-right-radius: 0;
            border-bottom-right-radius: 0;
            margin-right: -.5rem;
          }

          .react-markdown {
            border-top-left-radius: 0;
            border-bottom-left-radius: 0;
            padding: 1rem;
          }

          .fields-margin-fix {
            margin: 0 -.5em 1em !important;
          }

          .CodeMirror {
            height: 36rem;
          }

          .CodeMirror-scroll {
            padding-bottom: 0;
          }

          .center {
            text-align: center;
          }

          .underlined {
            border-bottom: 1px solid #ccc;
          }

          .CodeMirror-cursor {
            border-left: .5rem solid #ccc;
          }

        `}</style>
      </Layout>
    )
  }
}
const mapStateToProps = (state) => ({
  pendingTask: getPendingTask(state)
})

const mapDispatchToProps = dispatch => ({
  createTask: task => dispatch(createTask(task))
})

export default connect(mapStateToProps, mapDispatchToProps)(CreateTask)