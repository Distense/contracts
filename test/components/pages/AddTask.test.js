import React from 'react'
import sinon from 'sinon'
import { shallow, mount, render } from 'enzyme'
import { expect } from 'chai'

import {
  Button,
  Dropdown,
  Input,
  Form,
  Grid,
  Header,
  List,
  Message
} from 'semantic-ui-react'

import Head from '../../../src/components/common/Head'
import Layout from '../../../src/components/Layout'

import { AddTask } from '../../../src/pages/AddTask'

describe('<AddTask /> page component', function() {
  let wrapper
  let mounted
  const defaultProps = {
    title: '',
    tagsString: '',
    issueNum: '',
    numDIDRequiredToAddTask: 120,
    repo: '',
    redirect: false
  }

  beforeEach(() => {
    wrapper = shallow(<AddTask {...defaultProps} />)
  })

  it('should contain some basic subcomponents', function() {
    expect(wrapper.find(Layout).length).to.equal(1)
    expect(wrapper.find(Grid).length).to.equal(1)
    expect(wrapper.find(Head).length).to.equal(1)
    expect(wrapper.find(Grid.Row).length).to.equal(5)
    expect(wrapper.find(Grid.Column).length).to.equal(8)
    expect(wrapper.find(Form).length).to.equal(2)
    expect(wrapper.find(Form.Field).length).to.equal(4)
    expect(wrapper.find(Dropdown).length).to.equal(2)
    expect(wrapper.find(List).length).to.equal(5)
    expect(wrapper.find(Message).length).to.equal(5)
    expect(wrapper.find(Message.Header).length).to.equal(5)
  })

  it('should set the initial state correctly', function() {
    expect(wrapper.state('title')).to.equal('')
    expect(wrapper.state('tagsString')).to.equal('')
    expect(wrapper.state('issueNum')).to.equal('')
    expect(wrapper.state('numDIDRequiredToAddTask')).to.equal(120)
    expect(wrapper.state('redirect')).to.equal(false)
  })

  it('should contain a button with text "Submit"', function() {
    expect(wrapper.find(Button).length).to.equal(1)
  })
})
