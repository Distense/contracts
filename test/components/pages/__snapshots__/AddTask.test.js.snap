// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTask /> page component snapshot 1`] = `
<div>
  <div
    className="ui grid"
  >
    <div
      className="sixteen wide column"
    >
      <h2
        className="ui header"
      >
        Propose
      </h2>
      <p
        style={
          Object {
            "fontSize": "1.3rem",
          }
        }
      >
        DID holders can propose new work
      </p>
    </div>
    <div
      className="two column row"
      style={
        Object {
          "paddingTop": "0px",
        }
      }
    >
      <div
        className="column"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <div
              className="ui input"
            >
              <input
                disabled={undefined}
                onChange={[Function]}
                placeholder="Github Issue Number"
                tabIndex={undefined}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="field"
          >
            <div
              className="ui input"
            >
              <input
                disabled={undefined}
                name="title"
                onChange={[Function]}
                placeholder="Title (<70 chars)"
                tabIndex={undefined}
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            className="ui green large button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
      <div
        className="eight wide column"
      >
        <form
          action={undefined}
          className="ui form"
          onSubmit={[Function]}
        >
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid multiple search selection scrolling dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <span
                className="sizer"
              />
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                Tags
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={true}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  num="0"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Contracts
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="1"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    DApp Proposal
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="2"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Design
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="3"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Frontend
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="4"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Governance
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="5"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Parameters
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="6"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Planning
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="7"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    React
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  num="8"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Security
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="field"
          >
            <div
              aria-busy={undefined}
              aria-disabled={undefined}
              aria-expanded={false}
              className="ui fluid search selection scrolling dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
              tabIndex={undefined}
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                className="default text"
                role="alert"
              >
                Repo
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                aria-multiselectable={undefined}
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={true}
                  className="selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    UI
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-disabled={undefined}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    Contracts
                  </span>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="sixteen wide column"
      >
        <div
          className="ui message"
        >
          <div
            className="header"
          >
            Rules
          </div>
          <div
            className="ui bulleted list"
            role="list"
          >
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              You must own at least 
              120
               DID to propose. This number changes according to the
               
              <a
                href="/parameters"
                onClick={[Function]}
              >
                proposalPctDIDToApprove
              </a>
               
              parameter
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Your proposal can be anything, it doesn't necessarily have to be 
              <em>
                work
              </em>
              .
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Remember that when you propose, it is likely that DID will be issued for the completion of the task. The fewer DID the better.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="eight wide column"
      >
        <div
          className="ui message"
        >
          <div
            className="header"
          >
            Github Issue Number
          </div>
          <div
            className="ui bulleted list"
            role="list"
          >
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Create an issue in the appropriate Github repository to determine the Github issue number
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Your proposal can be anything, it doesn't necessarily have to be 
              <em>
                work
              </em>
              .
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Discussion will happen on Github in the issue you create
            </div>
          </div>
        </div>
      </div>
      <div
        className="eight wide column"
      >
        <div
          className="ui message"
        >
          <div
            className="header"
          >
            Title
          </div>
          <div
            className="ui bulleted list"
            role="list"
          >
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Enter the title of your task/proposal
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Limit the length of the title: the more characters the more gas it will cost you
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Example: 'Build this amazing new dapp that will change the world'
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              People will like you more if you enter the same exact title as your Github issue
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="eight wide column"
      >
        <div
          className="ui message"
        >
          <div
            className="header"
          >
            Tags
          </div>
          <div
            className="ui bulleted list"
            role="list"
          >
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Select the category of proposal you're making
            </div>
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Examples are: 'contracts', 'frontend', 'governance'
            </div>
          </div>
        </div>
      </div>
      <div
        className="eight wide column"
      >
        <div
          className="ui message"
        >
          <div
            className="header"
          >
            Repo
          </div>
          <div
            className="ui bulleted list"
            role="list"
          >
            <div
              className="item"
              data-value={undefined}
              onClick={[Function]}
              role="listitem"
            >
              Choose the one that best fits
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
